internal/buffer/settings.go

```diff
@@ -13,7 +15,12 @@ func (b *Buffer) SetOptionNative(option string, nativeValue interface{}) error {
 			if !b.Modified() {
 				e := calcHash(b, &b.origHash)
 				if e == ErrFileTooLarge {
-					b.Settings["fastdirty"] = false
+					b.Settings["fastdirty"] = true
+				}
+			} else {
+				b.origHash = [md5.Size]byte{}
+				if b.Size() > LargeFileThreshold {
```

Comment:
Just FYI this made me reveal a bunch of other small issues around `calcHash()`: #3430

...And just now I've also realized that this checking of file size inside `calcHash()` itself doesn't seem to make much sense, actually. It doesn't really optimize anything, since just calculating the file size (i.e. adding up line lengths) is probably faster than md5 calculation in any case (even if we optimized `calcHash()` to break the loop once `size` exceeds `LargeFileThreshold`).

So it seems it would be both simpler, cleaner and faster to let `calcHash()` just unconditionally calculate the hash of the given buffer, and explicitly check its size before calling `calcHash()` in all cases when size checking is needed.

Reply:
> So it seems it would be both simpler, cleaner and faster to let `calcHash()` just unconditionally calculate the hash of the given buffer, and explicitly check its size before calling `calcHash()` in all cases when size checking is needed.

Indeed, good point. Then `ErrFileTooLarge` becomes superfluous and buffers `Size()` compared against the `LargeFileThreshold` will be the only condition to decide if the hashing shall be performed.

Most probably it's then better to remove the commit https://github.com/zyedidia/micro/pull/3343/commits/61a76661996874d13ab171be1c6e03245dc5adb7 from this PR and move the changes to yours?

Reply:
> Most probably it's then better to remove the commit [61a7666](https://github.com/zyedidia/micro/commit/61a76661996874d13ab171be1c6e03245dc5adb7) from this PR and move the changes to yours?

I'm fine either way. Yeah, I can take care of that in my PR.

Reply:
Done. I added your https://github.com/zyedidia/micro/pull/3430/commits/352fd2be223bd1b12adb8abf3dc74651c89f9394 to #3430.

Also among other things I found one more issue here, fixed by https://github.com/zyedidia/micro/pull/3430/commits/93efc9eabe23adec1a33049470dced484f0d1630.